plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/messages.properties

```diff
@@ -624,8 +624,8 @@ EXP_CAST_FAILED=error;;;Cannot cast from {0} to {1}
 EXP_CAST_UNNECESSARY=warning;;;Unnecessary cast from {0} to {1}
 # no parameter required
 EXP_CAST_INVALID_TARGET=error;;;Can only cast to class, interface, enum, function, primitive, union or intersection types
-# 0: recognized type
-EXP_IS_NOT_A_VALID_NUMBER=error;;;{0} is not a valid number.
+# 0: recognized type, 1: expression type
+EXP_RESULT_IS_NAN=warning;;;Operand of type {0} in {1} will always result in NaN.
```

Comment:
The actual reason for getting `NaN` is, in most cases, not the type of the operand but the actual value of the operand. Even if those "most cases" are not implemented yet, I'd rephrase the message to:
```
An operand of {0} in {1} will always result in NaN.
```

Reply:
Updated the message template, but the message will remain the same since the validation stems from a type check.

We will probably replace these checks with data-analysis related checks anyway.